src/main/java/school/hei/haapi/authentication/AwsCognitoJwtAuthFilter.java

```diff
@@ -0,0 +1,37 @@
+package school.hei.haapi.authentication;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+@Component
+public class AwsCognitoJwtAuthFilter extends GenericFilter {
+
+    private static final Log logger = LogFactory.getLog(AwsCognitoJwtAuthFilter.class);
+    @Autowired
+    private AwsCognitoJwtIdToken cognitoIdTokenProcessor;
```

Comment:
1. No setter/attribute injection. Use constructor injection https://dzone.com/articles/spring-di-patterns-the-good-the-bad-and-the-ugly
2. Use final attributes as much as possible (and more genarally, immutable value)

Reply:
Configuration's attributes are now constant (static final) and beans are injected by constructor